VSConfigFinder/Program.cs

```diff
@@ -23,36 +23,34 @@ public static void Main(string[] args)
             {
                 with.CaseSensitive = false;
             });
-
-            // Take in the command line arguments
+            
             parser.ParseArguments<CommandLineOptions>(args)
                 .WithParsed(Run)
                 .WithNotParsed(HandleParseError);
         }
 
         private static void Run(CommandLineOptions options)
         {
-            // Run
-            if (options.CreateFile)
-            {
-                options.ConfigOutputPath ??= Directory.GetCurrentDirectory();
-            }
+            var fileSystem = new FileSystem();
+
+            ResolveCommandLineOptions(options);
 
-            // TODO: Replace this with the output from the runner
             var finalConfig = new VSConfig()
             {
+                // This is the new final .vsconfig, so version 1.0 is used.
                 Version = new Version("1.0"),
-                Components = new string[]
-                {
-                    "Microsoft.VisualStudio.Component.NuGet",
-                    "Microsoft.VisualStudio.Component.Roslyn.Compiler",
-                    "Microsoft.Component.MSBuild",
-                    "Microsoft.NetCore.Component.Runtime.6.0",
-                },
+                Components = Utilities.ReadComponents(fileSystem, options),
             };
 
-            // Output
-            Utilities.CreateOutput(finalConfig, options);
+            Utilities.CreateOutput(fileSystem, finalConfig, options);
```

Comment:
nit: why not just pass the components here and generate the vsconfig in here, if needed based on the cmdline switches?

Reply:
Mostly just for testability, I wanted to move most of the logic out of `program.cs`.